src/providers/fastly/runtime.ts

```diff
@@ -16,10 +16,21 @@ const operationsGenerator = createOperationsGenerator({
 })
 
 export default <RuntimeProvider> {
-  generateURL (src: string, modifiers: ImageModifiers, options: any) {
+  getImage (src: string, modifiers: ImageModifiers, options: any) {
     const operations = operationsGenerator(modifiers)
+    const url = cleanDoubleSlashes(options.baseURL + src + '?' + operations)
     return {
-      url: cleanDoubleSlashes(options.baseURL + src + '?' + operations)
+      url,
+      getInfo: async () => {
+        const infoString = await fetch(url).then(res => res.headers.get('fastly-io-info') || '')
```

Comment:
A tip for later: We may introduce a fetch utility (imported from `~image/fetch`) that can also do an in memory LRU cache